### ISSUE TITLE
rollbar.js floods with error

### ISSUE THREAD

**Issue by MaximKraev** [rollbar/rollbar.js#486]
We are getting strange behaviour then we get a error occurrence reports about a error 60 times per minute and never stops, there is no way to stop it except blocking IP what we are currently doing,
In sept we had 23,649 error occurrences, now it is about 10 times that. Support can't help as and sends here, any idea how we can resolve that?

**Comment by rokob** [comment]
Have you tried using the `maxItems` and `itemsPerMinute` settings?

**Comment by MaximKraev** [comment]
this is not a solution, it just workaround we will be changed for anyway.

**Comment by MaximKraev** [comment]
as I understand we hit exactly` itemsPerMinute` default rate and `maxItems` can filter real errors from client

**Comment by rokob** [comment]
1. Those rate limits are on the client side, so you won't be charged for anything beyond those limits.
2. If there is an error in your code generating this many errors, you can use the `checkIgnore` function to filter out those specific errors.
3. If you believe this is a bug in rollbar.js then can you provide more details so I can debug?

**Comment by MaximKraev** [comment]
1. Yes I know, but 60 items per minute is a limit currently, which is 86400 occurrences a day we are changed for
2. the error appears even before our code is loaded, it appears for different errors, some of them can't trigger such flood at all.
3. Could you please point me where I can get information you need?

**Comment by MaximKraev** [comment]
Anything I can help you with to diagnose the issue?

**Comment by simlu** [comment]
@rokob Which version does this happen with?

**Comment by MaximKraev** [comment]
we are using 2.2.8 currently

**Comment by MaximKraev** [comment]
we are using 2.2.8 currently

**Comment by aharpervc** [comment]
What's the nature of the reports? I saw something like this, possibly due to unknown chrome extension behavior. I filter them out with `checkIgnore`:

```js
checkIgnore: function(isUncaught, args, payload) {
            // extremely conservative rejection for only the most obvious, high volume false positives
            return (isUncaught
                && args === undefined
                && payload.body.trace.exception.class == "(unknown)"
                && payload.body.trace.exception.description == "unhandled rejection was null or undefined!"
                && payload.body.trace.frames.length == 1
                && payload.body.trace.frames[0].filename == "(unknown)"
                && payload.body.trace.frames[0].method == "[anonymous]");
        },
```

**Comment by MaximKraev** [comment]
we have filter for non site domains, also it happens for any errors. more often for "(unknown) Script error"

**Comment by rokob** [comment]
I'm mostly digging around in the source blind because there isn't much to go on here. We have code to limit sending the same error twice, but it only stops if we have an instance of Error. But for cases of uncaught errors this should be valid. I could change that to add a configurable option to not send anything that looks like a duplicate of the last item including messages. It would have to be configurable because that usually is not what you want.

Hence, most times when there are reports of multiple instances of the same error, it is usually because some script on the page (sometimes injected via a browser plugin) is caught in a loop and repeatedly throwing errors.

**Comment by jessewgibbs** [comment]
Related discussion in Intercom: https://app.intercom.io/a/apps/rlisg82q/respond/inbox/mentions/conversations/12148287229

**Comment by jessewgibbs** [comment]
Related discussion in Intercom: https://app.intercom.io/a/apps/rlisg82q/respond/inbox/mentions/conversations/12148287229

**Comment by MaximKraev** [comment]
It was not a repeated error in a loop, script error happed only once, rollbar has sent it 60.000 times, until i banned the client by IP,, it wasn't related to a plugin as we have filters by domins after and the same happened. You might be looking for an issue a wrong place, problem is not with the configuration, problem is a bug in the library.

**Comment by rivkahstandig3636** [comment]
This issue was solved in the Intercom conversation and is now closed.

**Comment by aharpervc** [comment]
> This issue was solved in the Intercom conversation and is now closed.

Well, what was the solution?

**Comment by v3rron** [comment]
We're having the same issue.
Upgraded our rollbar.js library from 1.9.4 to 2.3.1 and just got 10,000 errors:
```
(unknown): unhandled rejection was null or undefined!
File "(unknown)" line (unknown) col 0 in [anonymous]
#179 maxItems has been hit. Ignoring errors until reset.
```
It seems to be coming from the same IP address on various pages of our app. We didn't change our code for months, just upgraded the Rollbar library.

**Comment by MaximKraev** [comment]
> Well, what was the solution?

We migrated from rollbar.

**Comment by rokob** [comment]
This is a bit of a tough one. Looking at the occurrences associated with these errors, they have different client generated timestamps. Those timestamps only get generated in `createItem` calls which only happen when we handle an uncaught exception, an unhandled rejection, or a direct call to one of the logging methods.

Hence, while this might look like some type of bug where Rollbar is sending the same error repeatedly, that isn't possible based on the code paths without an external source of method calls into Rollbar (e.g. `unhandledrejection` event handler being called by the browser).

It is certainly possible for a script loading to generate an unbounded number of rejected promises. I am not saying there is not a bug in this library, but the evidence so far is pointing to something happening a level up.

One solution we can do would be to only send one instance of an error if it looks the same as a previous one even if it is generated by something external. There is some code currently that is intended to do that, but I think it is broken.

**Comment by claytonrcarter** [comment]
I would also like to know what the linked solution is. We're seeing similar problems, all related to Chrome. On 12/19/17 we had a single user generate almost of 94,000 identical errors in a matter of hours, all from the same 2 pages. (The error was "unhandled rejection was null or undefined!" in Chrome 62 on Windows 10.) This maxed out our quota and put us in the dark until our quota reset this morning. After our quota reset this morning, we started receiving hundreds of similar occurrences ("unhandled rejection was null or undefined!" in Chrome 63 on MacOS 10.11).

We are seeing these errors stack up from limited users (only 2, in fact) but they accumulate 1-2 dozen at a time, sometimes every second, sometimes an hour apart. Would love some more info on tackling this, or at least limiting it.

We updated all of our script links in case it was a CORS related problem, but that doesn't seem to have fixed anything.
